### PR TITLE
Remove celery

### PR DESCRIPTION
**Pull request by scanon**
This PR replaces celery with standard python multiprocessing threads.  This should simplify installation and operation.

### CODE REVIEW

**Comment by scanon** [comment]
This coverage drop is misleading.  I'm now excluding the test directory from the coverage report which was inflating things a bit.

File: doc/faq.rst [hunk 1]
```diff
@@ -15,14 +15,12 @@ system or be installed in trusted locations.
...
+Yes.  If you are seeing errors similar to the following in the image gateway log
 then you'll need to include the provide sitecustomize.py in your python setup
-or just point the celery process's PYTHONPATH to include it::
+or just point the gateway's PYTHONPATH to include it::
 
     [2016-08-01 09:41:20,664: ERROR/MainProcess] Task shifter_imagegw.imageworker.dopull[XXXXXXX-omitted-XXXXXXX] raised unexpected: UnicodeDecodeError('ascii', '/path/is/omitted/some\xc3\xa9_unicode', 35, 36, 'ordinal not in range(128)')
     Traceback (most recent call last):
-      File "/usr/lib/python2.6/site-packages/celery/app/trace.py", line 240, in trace_task
-          R = retval = fun(*args, **kwargs)
       File "/usr/lib/python2.6/site-packages/celery/app/trace.py", line 438, in __protected_call__
```
Comment (Madeeks) [inline doc/faq.rst]: The stack frame above (with an explicit reference to Celery) was removed, but this one remains.
Is there a specific reason?

File: imagegw/shifter_imagegw/imagemngr.py [hunk 2]
```diff
@@ -17,12 +17,14 @@
...
 # See LICENSE for full text.
 
-#
-# This module is provides the interface layer for the image manager.  This
-# compliments the api module which provides the REST interface.  This module
-# does much of the heavy lifting for the image manager.  It handles all
-# interactions with the Mongo Database and dispatches work through Celery.
-#
+"""
+Imagae Manager for the Shifter Gateway.
```
Comment (Madeeks) [inline imagegw/shifter_imagegw/imagemngr.py]: Typo here.

File: imagegw/shifter_imagegw/imagemngr.py [hunk 3]
```diff
@@ -101,19 +99,58 @@ def __init__(self, config, logger=None, logname='imagemngr'):
...
+    def shutdown(self):
+        self.status_queue.put('stop')
+
+    def status_thread(self):
+        """
+        This listens for update messages from a queue.
+        """
+        client = MongoClient(self.config['MongoDBURI'])
+        db_ = self.config['MongoDB']
+        self.images = client[db_].images
```
Comment (Madeeks) [inline imagegw/shifter_imagegw/imagemngr.py]: Lines 124-126 are the same as 109-111. You could consider adding a dedicated function to remove the duplication.

File: imagegw/shifter_imagegw/imageworker.py [hunk 4]
```diff
@@ -54,42 +53,68 @@
...
+        try:
+            remove_image(request, updater)
+        except Exception as err:
+            resp = {'error_type': str(type(err)),
+                    'message': str(err)}
+            updater.update_status('FAILURE', 'FAILURE', response=resp)
+
+    def dopull(self, ident, request, testmode=0):
+        """
+        Kick off a pulll operation.
```
Comment (Madeeks) [inline imagegw/shifter_imagegw/imageworker.py]: Typo here.

**Pull request merged**